diff -Naur a/kernel-open/nvidia/nv.c b/kernel-open/nvidia/nv.c
--- a/kernel-open/nvidia/nv.c	2026-07-18 11:26:26.928006389 -0700
+++ b/kernel-open/nvidia/nv.c	2026-07-18 11:26:26.930357872 -0700
@@ -1521,6 +1521,15 @@
         goto failed_release_irq;
     }
 
+    if (nv->pci_info.device_id == 0x20C2)
+    {
+        nv->flags |= NV_FLAG_PERSISTENT_SW_STATE;
+    }
+    else if (nv->pci_info.device_id == 0x2082)
+    {
+        nv->flags |= NV_FLAG_PERSISTENT_SW_STATE;
+    }
+
     /* Generate and cache the UUID for future callers */
     (void)rm_get_gpu_uuid_raw(sp, nv);
 
